renku/core/commands/providers/renku.py

```diff
@@ -319,15 +285,56 @@ def _extract_dataset_id(uri):
     @property
     def version(self):
         """Get record version."""
-        return self._dataset.version
+        return self._dataset.version if self._dataset else None
 
     @property
     def latest_uri(self):
-        """Get uri of latest version."""
-        return self._dataset._id
+        """Get uri of the latest version."""
+        return self._latest_version_uri
 
     @property
     def datadir_exists(self):
         """Whether the dataset datadir exists (might be missing in git if empty)."""
+        return (self._remote_client.path / self._dataset.data_dir).exists()
+
+    def _fetch_dataset(self, client):
+        repo_path = None
+
+        urls = (self._project_url_ssh, self._project_url_http)
+        # Clone the project
+        for url in urls:
+            try:
+                repo, repo_path = client.prepare_git_repo(url)
+            except errors.GitError:
+                pass
+            else:
+                self._project_url = url
+                break
+
+        if not self._project_url:
+            raise errors.ParameterError("Cannot clone remote projects:\n\t" + "\n\t".join(urls), param_hint=self._uri)
+
+        self._remote_client = LocalClient(repo_path)
+        self._migrate_project(self._remote_client)
+
+        self._dataset = self._remote_client.load_dataset(self._name)
+
+        if not self._dataset:
+            raise errors.ParameterError(f"Cannot find dataset '{self._name}' in project '{self._project_url}'")
+
+        for file_ in self._dataset.files:
+            file_.checksum = self._remote_client.repo.git.hash_object(file_.path)
+            file_.filesize = self._get_file_size(self._remote_client, file_.path)
```

Comment:
```suggestion
            file_.filesize = _RenkuRecordSerializer._get_file_size(self._remote_client, file_.path)
```

since it's a static method now